src/cautiousrobot/__main__.py

```diff
@@ -103,7 +105,7 @@ def download_images(data, img_dir, log_filepath, error_log_filepath, filename =
             max_redos = retry
             while redo and max_redos > 0:
                 try:
-                    response = requests.get(url)
+                    response = requests.get(url, stream = True)
```

Comment:
When the input CSV doesn't have a checksum column we might fail to download the entire file without notifying the user. Could you check the [response headers](https://requests.readthedocs.io/en/latest/user/quickstart/#response-headers) for 
[Content-Length](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Length) and compare that value to the size of the full size image file? Often web servers will set this response header so it would be nice to check it when this value is sent in the response.

Reply:
Looks like I could do the following?
```python
img_size = response.headers["Content-Length"]
if img_size != os.stat('response.raw').st_size: # os.stat from https://stackoverflow.com/a/11904141
  redo = True
  max_redos -= 1
  if max_redos <= 0:
    log_errors = log_response(log_errors,
                                        index = i,
                                        image = image_name,
                                        url = url,
                                        response_code = "200-incomplete")
                        update_log(log = log_errors, index = i, filepath = error_log_filepath)
  continue
```

Reply:
This does significantly slow it down:
```
cautious-robot --input-file examples/HCGSD_testNA.csv --output-dir examples/test_images
100%|██████████████████████████████████████████████| 8/8 [00:07<00:00,  1.13it/s]
```
compared to
```
cautious-robot --input-file examples/HCGSD_testNA.csv --output-dir examples/test_images
100%|██████████████████████████████████████████████| 8/8 [00:01<00:00,  5.60it/s]
```
I think maybe we set this as an issue (tied to the potential retry logic change). I could add expected image size to the logging for now, but the immediate use-cases of the alpha version will have MD5s to compare their downloads.

Reply:
I wouldn't expect `os.stat()` to slow anything down like this. Is this comparing streaming vs not streaming as well?

Reply:
Let's move checking the file size to another PR for simplicity.

Reply:
Yeah, it was just changing those 20 lines (slight reordering to implement the `img_size` check).